### PR TITLE
Move to using unstructured.Unstructured rather than extv1.CustomResourceDefinition

### PR DESCRIPTION
**Pull request by tnthornton**
### Description of your changes

This is a follow on to https://github.com/upbound/xgql/pull/114.

With the changes from #114, we saw that following resource utilization graphs (notably much better than prior to the change):
Screen Shot 2023-04-21 at 5 50 18 PM

However as seen in the heap pprof after the fix:

Screen Shot 2023-04-21 at 5 54 44 PM

There's memory consumption occurring in `JSONSchemaProps*`. As referenced in #114, we've seen high utilization when un/marshaling `CustomResourceDefinition`s; much of this comes from how expensive it is to process the `JSONSchemaProps*`. Fortunately, these pieces are necessary for producing the `CustomResource` Validators - which is unnecessary for `xgql`.

With that in mind, this PR moves to using `Unstructured` instead of the full `CustomResourceDefinition` types. With just that adjustment we see the following differences:

Heap pprof:
Screen Shot 2023-04-21 at 6 00 13 PM

Utilization graphs:
Screen Shot 2023-04-21 at 6 00 42 PM

That's great. However, we didn't stop there. While grabbing pprofs, I saw that we were still spending quite a bit of time marshaling types:
Screen Shot 2023-04-21 at 6 04 25 PM

Due to this, I moved to using the standard `encoding/json` package which had a positive impact, but then I saw a bit of time spent in `unstruct()`:
Screen Shot 2023-04-21 at 6 07 24 PM

Due to this, I moved to using json/v2 which was the approach implemented upstream in k8s-api-server for the 1.26 to help with marshaling types, which ended having additional positive impacts:
Screen Shot 2023-04-21 at 6 10 41 PM

Lastly, I saw that we were spending a bit of time in unstruct for our new `CustomResourceDefinition`s. In order to address that, I added a specialized method, which resulted in the final impacts on both heap and cpu:

Heap:
Screen Shot 2023-04-21 at 6 13 00 PM

CPU:
Screen Shot 2023-04-21 at 6 14 05 PM

Highlevel changes:
- [X] add new unstructured.CustomResourceDefintion
- [X] add new unstructured.CustomResourceDefinitionList
- [X] specify that we don't have the informers cache extv1.CustomResourceDefinitions
- [X] add new specifialized bytesForUnstructured to remove unnecessary overhead for Marshaling unstuctured objects that are simply being returned as []byte
- [X] move to using json/v2 to take advantage of better un/marshal performance similar to 1.27 in the api-server

I have:

- [X] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [X] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested
1. Ensured the artifacts built and tests passed (`make test build.all`)
2. Deployed the image into the environment referenced in #114 and verified queries still worked and data still rendered.

### CODE REVIEW

File: internal/graph/resolvers/managed.go [hunk 1]
```diff
@@ -74,7 +74,14 @@ func (r *managedResource) Definition(ctx context.Context, obj *model.ManagedReso
...
 
 	nn := types.NamespacedName{Name: fmt.Sprintf("%s.%s", name, gv.Group)}
-	in := &kextv1.CustomResourceDefinition{}
+	// in := &unstructured.CustomResourceDefinition{}
+	// xrc.SetAPIVersion(obj.ClaimReference.APIVersion)
+	// xrc.SetKind(obj.ClaimReference.Kind)
+	// apiVersion: apiextensions.k8s.io/v1
+	// kind: CustomResourceDefinition
+	// in.SetAPIVersion(crdAPIVersion)
+	// in.SetKind(crdKind)
```
Comment (thephred) [inline internal/graph/resolvers/managed.go:83]: Wanting to keep this?

File: internal/graph/resolvers/managed.go [hunk 2]
```diff
@@ -85,20 +92,23 @@ func (r *managedResource) Definition(ctx context.Context, obj *model.ManagedReso
 	// We didn't find the CRD we were looking for, list all CRDs and see if we
 	// can find the matching one.
 	if kerrors.IsNotFound(err) {
-		lin := &kextv1.CustomResourceDefinitionList{}
+		// lin := &kunstructured.UnstructuredList{}
+		// lin.SetAPIVersion(crdAPIVersion)
+		// lin.SetKind(crdKindList)
```
Comment (thephred) [inline internal/graph/resolvers/managed.go:97]: Wanting to keep this?

File: internal/graph/resolvers/provider.go [hunk 3]
```diff
@@ -173,7 +174,10 @@ func (r *providerRevisionStatus) Objects(ctx context.Context, obj *model.Provide
 			continue
 		}
 
-		crd := &kextv1.CustomResourceDefinition{}
+		// crd := &unstructured.CustomResourceDefinition{}
+		// crd.SetAPIVersion(crdAPIVersion)
+		// crd.SetKind(crdKind)
```
Comment (thephred) [inline internal/graph/resolvers/provider.go:179]: Wanting to keep this?

**Review by thephred — APPROVED** [review]
Looks good but wondering if we need to keep around some commented out code.

**Comment by tnthornton** [comment]
>Looks good but wondering if we need to keep around some commented out code.

Nice catch. Missed those..

**Pull request merged**